apps/src/templates/manageStudents/manageStudentsRedux.js

```diff
@@ -31,13 +35,41 @@ export default function manageStudents(state=initialState, action) {
       studentData: action.studentData,
     };
   }
+  if (action.type === START_EDITING_STUDENT) {
+    return {
+      ...state,
+      studentData: {
+        ...state.studentData,
+        [action.studentId]: {
+          ...state.studentData[action.studentId],
+          isEditing: true
+        }
+      }
+    };
```

Comment:
Note: An alternative to this would be to use Immutable.js, which I think would allow you to do something like
```js
return state.set(['studentData', action.studentId, isEditing'], true);
```

Doing some googling, I think you could also accomplish this with some lodashing, doing something like:
```js
return _.set(_.clone(state), `studentData.${action.studentId}.isEditing`, true);
```

I haven't validated that this works, and am mostly just sharing as an FYI. The pattern currently used here is one we use elsewhere

Reply:
👍

Reply:
Turns out this was bad advice. Clone only makes a shallow copy, so this still ends up doing a mutation.

Consider the following code:
```
var foo = {
  parent: {
    x: 1,
    y: 2
  }
};

console.log(foo.parent.x); // prints 1

var foo2 = _.set(_.clone(foo), 'parent.x', 3);

console.log(foo.parent.x); // prints 3, foo and _.clone(foo) are both using the same parent object
```

Reply:
The following does appear to work, but I'm not sure I grok it well enough to recommend it's usage:
```
var foo2 = _.setWith(_.clone(foo), 'parent.x', 3, _.clone)
```

Reply:
alright, but it sounds like they way it is currently works.

Reply:
Yep. Current approach is 👍